doc/developer/design/20210401_volatile_sources.md

```diff
@@ -0,0 +1,294 @@
+# Volatile sources
+
+## Summary
+
+Our users want to connect Materialize to upstream sources that do not provide
+guarantees about what data they will present. These sources are in tension with
+the goal of providing a deterministic, replayable system.
+
+This design doc proposes to:
+
+  * Introduce "volatile source" as a term of art that describes a source that
+    does not have upstream durability.
```

Comment:
It seems like several of the problems that this doc captures also arise if the source doesn't have a deterministic message delivery order.

Reply:
Ack. Reworded this to say "reliability" instead of limiting to durability.